conans/client/build/cmake_flags.py

```diff
@@ -19,10 +19,15 @@
 
 
 def get_toolset(settings):
-    if settings.get_safe("compiler") == "Visual Studio":
+    compiler = settings.get_safe("compiler")
+    compiler_base = settings.get_safe("compiler.base")
+    if compiler == "Visual Studio":
         subs_toolset = settings.get_safe("compiler.toolset")
         if subs_toolset:
             return subs_toolset
+    elif compiler == "intel" and compiler_base == "Visual Studio":
+        compiler_version = settings.get_safe("compiler.version")
+        return "Intel C++ Compiler " + compiler_version
```

Comment:
Need this to be (e.g.) "Intel C++ Compiler 19.0" not "Intel C++ Compiler 19". Also, if "compiler.version" is not set, this function should probably return `None`.

Reply:
This change also breaks when an explicit generator is set that is not Visual Studio project files. I think the issue the root cause is [here](https://github.com/SSE4/conan/blob/intel_toolset/conans/client/build/cmake.py#L177), which should add a `"Visual" in self.generator` check.

Reply:
addressed both problems, added an additional test, thanks for the catching